src/formatDate.js

```diff
@@ -50,7 +50,47 @@
  */
 
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const data = date
+    .split(fromFormat.slice(-1))
+    .reduce((dateInfo, datePart, i) => {
+      dateInfo[fromFormat[i][0]] = datePart;
+
+      return dateInfo;
+    }, {});
+
+  const newDate = [];
+  const separator = toFormat.slice(-1);
+
+  for (let i = 0; i < toFormat.length - 1; i++) {
+    const dateFormat = toFormat[i];
+    const datePart = data[dateFormat[0]];
+
+    newDate.push(dateFormat[0] === 'Y'
+      ? handleYears(dateFormat, datePart, fromFormat)
+      : datePart);
+  }
+
+  return newDate.join(separator);
+}
+
+/**
+ * @param {string} dateFormat
+ * @param {string} years
+ * @param {string[]} fromFormat
+ *
+ * @returns {string}
+ */
+
+function handleYears(dateFormat, years, fromFormat) {
+  if (dateFormat === 'YY') {
+    return years.slice(-2);
+  }
+
+  if (fromFormat.includes('YY')) {
+    return (+years < 30 ? '20' : '19') + years;
```

Comment:
Make a variable for your ternary, with proper formatting, too)

Reply:
Do I need to do something like that

![image](https://user-images.githubusercontent.com/72860434/141278802-0be89f10-7f26-4a90-9716-d8aa1a502b69.png)

or since this is short, I can put it in one line? Like that =>

![image](https://user-images.githubusercontent.com/72860434/141278925-cc3aad9e-920f-458f-8492-e6ac84d4630c.png)

Reply:
First option looks great